src/main/java/teammates/common/util/Config.java

```diff
@@ -95,7 +95,11 @@ private String getGcsBucketname() {
     }
 
     private String getAppUrl() {
-        return props.getProperty("app.url");
+        return Url.trimTrailingSlash(props.getProperty("app.url"));
+    }
+    
+    public static AppUrl getAppUrl(String relativeUrl) {
```

Comment:
Yes, widespread change is going to make the PR hard to manage, and I prefer to do those small but widespread refactorings as separate independent PRs. On a more general note, needing widespread change is an incentive to do it rather than the opposite. Widespread change is better than widespread code smells. The more widespread and more numerous something is, the more reason to keep it very very clean.